linode/manager
# Use `swc` to run Jest tests

## Description

**What does this PR do?**
This switches the transformer used to transpile our Jest tests from `ts-jest` to [`@swc/jest`](https://swc.rs/docs/usage/jest). I additionally included the [`jest_workaround`](https://www.npmjs.com/package/jest_workaround) plugin which allows us to mutate ES modules -- something that `ts-jest` allows but `@swc/jest` does not, and which is required for `jest.spyOn(...)` to work.

On my machine, this reduces manager's (uncached) Jest run time from about **8 minutes** to around **100 seconds**.

## How to test

**What are the steps to reproduce the issue or verify the changes?**

Confirm that tests work as expected for manager and api-v4:
1. Run `yarn` to install the new dependencies
2. Run `yarn test --clearCache` to clear Jest's cache
    * This is not strictly necessary, but makes it easier to compare Jest run time between this branch and `develop`
3. Run `yarn test`, observe that tests pass and performance is improved
